palette/moreland/convert.go

```diff
@@ -0,0 +1,228 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package moreland
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+)
+
+// linearToS converts from a linear RGB component to an sRGB component.
+func linearToS(v float64) float64 {
+	if v > 0.0031308 {
+		return 1.055*math.Pow(v, 1/2.4) - 0.055
+	}
+	return 12.92 * v
+}
+
+// sToLinear converts from an sRGB component to a linear RGB component.
+func sToLinear(v float64) float64 {
+	if v > 0.04045 {
+		return math.Pow((v+0.055)/1.055, 2.4)
+	}
+	return v / 12.92
+}
+
+// linearRGB repesents a physically linear RGB color.
+type linearRGB struct {
+	R, G, B float64
+}
+
+// XYZ converts a linear RGB color to a CIE XYZ color.
+func (c linearRGB) XYZ() cieXYZ {
+	return cieXYZ{
+		X: 0.4124*c.R + 0.3576*c.G + 0.1805*c.B,
+		Y: 0.2126*c.R + 0.7152*c.G + 0.0722*c.B,
+		Z: 0.0193*c.R + 0.1192*c.G + 0.9505*c.B,
+	}
+}
+
+// linearRGB converts a CIEXYZ color to physically linear RGB.
+func (c cieXYZ) linearRGB() linearRGB {
+	return linearRGB{
+		R: c.X*3.2406 + c.Y*-1.5372 + c.Z*-0.4986,
+		G: c.X*-0.9689 + c.Y*1.8758 + c.Z*0.0415,
+		B: c.X*0.0557 + c.Y*-0.204 + c.Z*1.057,
+	}
+}
+
+// labTemp is an intermediate step in converting from CIE XYZ to CIE LAB.
```

Comment:
Is there a better name than "labTemp"? (temp in the context of a colour package bring temperature to mind).

Move it below its use.